packages/node_modules/@webex/plugin-meetings/src/meeting/index.js

```diff
@@ -2648,39 +2648,40 @@ export default class Meeting extends StatelessWebexPlugin {
    * @public
    * @memberof Meeting
    */
-  stopShare() {
-    const content = this.locusInfo.mediaShares.find((element) => element.name === CONTENT);
+  async stopShare() {
+    const content = this.locusInfo.mediaShares
+      .find((element) => element.name === CONTENT);
 
     if (content && this.isSharing) {
-      Metrics.postEvent({event: eventType.SHARE_STOPPED, meeting: this});
-      Media.stopTracks(this.mediaProperties.shareTrack);
-
-      return this.meetingRequest.changeMeetingFloor({
-        disposition: FLOOR_ACTION.RELEASED,
-        personUrl: this.locusInfo.self.url,
-        deviceUrl: this.deviceUrl,
-        uri: content.url,
-        resourceUrl: this.resourceUrl
-      })
-        .catch((e) => {
-          LoggerProxy.logger.error('Meeting/index->stopShare#Error ', e);
+      try {
+        Metrics.postEvent({event: eventType.SHARE_STOPPED, meeting: this});
+        Media.stopTracks(this.mediaProperties.shareTrack);
 
-          return Promise.reject(e);
-        })
-        .finally(() => {
-          this.isSharing = false;
-          Trigger.trigger(
-            this,
-            {
-              file: 'meeting/index',
-              function: 'stopShare'
-            },
-            EVENT_TRIGGERS.MEETING_STOPPED_SHARING_LOCAL,
-          );
+        await this.meetingRequest.changeMeetingFloor({
+          disposition: FLOOR_ACTION.RELEASED,
+          personUrl: this.locusInfo.self.url,
+          deviceUrl: this.deviceUrl,
+          uri: content.url,
+          resourceUrl: this.resourceUrl
         });
+      }
+      catch (error) {
+        LoggerProxy.logger.error('Meeting/index->stopShare#Error ', error);
+        throw error;
+      }
+
+      this.isSharing = false;
```

Comment:
Same here, can we do a try catch finally?

Also shouldn't we still return a promise resolution/rejection as this is a user interaction call

Reply:
I will see if angular has the same issues with finally. This function didn’t return anything on the resolve, so it will behave the same due to it being asynchronous. I switched the promise rejection to a throw for the same reason